integrationTests/ts/__benchmarks__/benchmarks.ts

```diff
@@ -0,0 +1,367 @@
+import benny from "benny";
+import { genProof } from "maci-circuits";
+import { CircuitInputs, MaciState } from "maci-core";
+import { Keypair, PCommand } from "maci-domainobjs";
+
+import { homedir } from "os";
+import path from "path";
+
+import { STATE_TREE_DEPTH, duration, maxValues, treeDepths, voiceCreditBalance } from "../__tests__/utils/constants";
+
+const NAME = "proof generation";
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export default async function rune2e(): Promise<void> {
+  const coordinatorKeypair = new Keypair();
+
+  const messageBatchSize = 5;
+
+  const users = new Array<Keypair>(5).fill(new Keypair());
+
+  benny.suite(
+    NAME,
+
+    benny.add(`wasm genProof - 5 signups, 5 messages`, async () => {
+      const maciState = new MaciState(STATE_TREE_DEPTH);
+      const pollId = maciState.deployPoll(
+        BigInt(duration),
+        maxValues,
+        treeDepths,
+        messageBatchSize,
+        coordinatorKeypair,
+      );
+      const poll = maciState.polls.get(pollId);
+
+      // signup 5 users
+      // eslint-disable-next-line @typescript-eslint/prefer-for-of
+      for (let i = 0; i < users.length; i += 1) {
+        const keypair = users[i];
+        maciState.signUp(keypair.pubKey, voiceCreditBalance, BigInt(Date.now()));
+        // First command (valid)
+        const command = new PCommand(
+          BigInt(i) + 1n,
+          keypair.pubKey,
+          5n, // voteOptionIndex,
+          7n, // vote weight
+          2n, // nonce
+          pollId,
+        );
+
+        const signature = command.sign(keypair.privKey);
+
+        const ecdhKeypair = new Keypair();
+        const sharedKey = Keypair.genEcdhSharedKey(ecdhKeypair.privKey, coordinatorKeypair.pubKey);
+        const message = command.encrypt(signature, sharedKey);
+        poll?.publishMessage(message, ecdhKeypair.pubKey);
+      }
+
+      while (poll?.hasUnprocessedMessages()) {
+        const pmInputs = poll.processMessages(pollId) as unknown as CircuitInputs;
+
+        // eslint-disable-next-line no-await-in-loop
+        await genProof({
+          inputs: pmInputs,
+          useWasm: true,
+          wasmPath: path.resolve(
+            __dirname,
+            "../../../cli/zkeys/ProcessMessages_10-2-1-2_test/ProcessMessages_10-2-1-2_test_js/ProcessMessages_10-2-1-2_test.wasm",
+          ),
+          zkeyPath: path.resolve(
+            __dirname,
+            "../../../cli/zkeys/ProcessMessages_10-2-1-2_test/ProcessMessages_10-2-1-2_test.0.zkey",
+          ),
+        });
+      }
```

Comment:
This code seems is duplicated. Let's move it to a separate function and reuse for all the benchmarks.

Reply:
sure